From b46c05a08ea806b899a87c24f50ee31bc36346c4 Mon Sep 17 00:00:00 2001
From: "Guillermo A. Nunez" <guillermo@cozybit.com>
Date: Tue, 8 May 2012 19:46:42 -0700
Subject: [PATCH 1/3] mac80211: specify the channel type when setting the
 channel.

This is needed to allow HT mode in mesh ifaces.

Signed-off-by: Guillermo A. Nunez <guillermo@cozybit.com>
---
 package/mac80211/files/lib/wifi/mac80211.sh |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/package/mac80211/files/lib/wifi/mac80211.sh b/package/mac80211/files/lib/wifi/mac80211.sh
index 8bb9c42..461b248 100644
--- a/package/mac80211/files/lib/wifi/mac80211.sh
+++ b/package/mac80211/files/lib/wifi/mac80211.sh
@@ -289,6 +289,7 @@ mac80211_generate_mac() {
 enable_mac80211() {
 	local device="$1"
 	config_get channel "$device" channel
+	config_get htmode "$device" htmode
 	config_get vifs "$device" vifs
 	config_get txpower "$device" txpower
 	config_get country "$device" country
@@ -393,7 +394,7 @@ enable_mac80211() {
 			# We attempt to set the channel for all interfaces, although
 			# mac80211 may not support it or the driver might not yet
 			# for ap mode this is handled by hostapd
-			[ -n "$fixed" -a -n "$channel" ] && iw dev "$ifname" set channel "$channel"
+			[ -n "$fixed" -a -n "$channel" ] && iw dev "$ifname" set channel "$channel" $htmode
 		fi
 
 		i=$(($i + 1))
-- 
1.7.5.4

